app/actions/app/server.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Navigation} from 'react-native-navigation';
+
+import {doPing} from '@actions/remote/general';
+import {fetchConfigAndLicense} from '@actions/remote/systems';
+import {Screens} from '@constants';
+import DatabaseManager from '@database/manager';
+import SecurityManager from '@managers/security_manager';
+import WebsocketManager from '@managers/websocket_manager';
+import {getServer, getServerByIdentifier, queryAllActiveServers} from '@queries/app/servers';
+import {getConfig} from '@queries/servers/system';
+import {logError} from '@utils/log';
+import {canReceiveNotifications} from '@utils/push_proxy';
+import {alertServerAlreadyConnected, alertServerError, loginToServer} from '@utils/server';
+
+import type {IntlShape} from 'react-intl';
+
+export async function initializeSecurityManager() {
+    const servers = await queryAllActiveServers()?.fetch();
+    if (!servers?.length) {
+        return;
+    }
+
+    const promises: Array<Promise<[string, ClientConfig | null]>> = [];
+    const results: Record<string, ClientConfig> = {};
+
+    for (const server of servers) {
+        try {
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(server.url);
+            const promise = getConfig(database).then((config) => [server.url, config] as [string, ClientConfig | null]);
+            promises.push(promise);
+        } catch (error) {
+            logError('initializeSecurityManager', error);
+            continue;
+        }
+    }
+
+    const resolvedConfigs = await Promise.allSettled(promises);
+
+    for (const result of resolvedConfigs) {
+        if (result.status === 'fulfilled') {
+            const [url, config] = result.value;
+            if (config && Object.keys(config).length > 0) { // Ensure the object is not empty
+                results[url] = config;
+            }
+        }
+    }
+
+    const serverUrl = await DatabaseManager.getActiveServerUrl();
+    SecurityManager.init(results, serverUrl);
+}
+
+export async function switchToServer(serverUrl: string, theme: Theme, intl: IntlShape, callback?: () => void) {
+    const server = await getServer(serverUrl);
+    if (!server) {
+        logError(`Switch to Server with url ${serverUrl} not found`);
+        return;
+    }
+    if (server.lastActiveAt) {
```

Comment:
Is this for allowing at least logging into the server from a jailbroken device? Can you please add a comment explaining the purpose of this check?

Reply:
server.lastActiveAt means that is an existing session on that server and is not a server that was logged out.

for example, if you have two servers in your app but logout from one of them and you want to connect to it again, it will return `lastActiveAt` as 0 meaning it does not hit this condition, if the server has an active session, then it will perform the necessary checks before allowing you to switch to it.

Reply:
Got it.